src/main/java/org/opensearch/timeseries/constant/CommonMessages.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.timeseries.constant;
+
+import java.util.Locale;
+
+public class CommonMessages {
+    // ======================================
+    // Validation message
+    // ======================================
+    public static String NEGATIVE_TIME_CONFIGURATION = "should be non-negative";
+    public static String INVALID_RESULT_INDEX_MAPPING = "Result index mapping is not correct for index: ";
+    public static String EMPTY_NAME = "name should be set";
+    public static String NULL_TIME_FIELD = "Time field should be set";
+    public static String EMPTY_INDICES = "Indices should be set";
+
+    public static String getTooManyCategoricalFieldErr(int limit) {
+        return String.format(Locale.ROOT, CommonMessages.TOO_MANY_CATEGORICAL_FIELD_ERR_MSG_FORMAT, limit);
+    }
+
+    public static final String TOO_MANY_CATEGORICAL_FIELD_ERR_MSG_FORMAT = "We can have only %d categorical field/s.";
```

Comment:
Not a nit. "Currently we only support up to %d categorical ..." -- may be even have "in order to bound system resource consumption." That would explain that there is no magic stopping this from happening. This should empower folks who have pressing needs to voice it.